### DISCUSSION TITLE
How do we contribute ourselves to this library?

### DISCUSSION THREAD

**Discussion by Md-Anamul-Haque** [Sec-ant/zxing-wasm#174] [Q&A] [answered]
How do we contribute ourselves to this library?
Please provide a setup guide.

**Comment by Sec-ant** [comment]
Hi, thanks for the offer. Do you have a specific feature or issue in your mind? This repo is currently maintained by myself alone in my free time. Maintaining an up-to-date *contribution guide* is another burden I'd love to avoid. That said, the code should be explanatory enough and the build step is written in the beginning of `README`. I'd also be glad to answer questions.

You can also check the github actions & workflow files to get an idea on how to set up the environment.

**Comment by Md-Anamul-Haque** [comment]
thanks

**Comment by Md-Anamul-Haque** [comment]
Please make something easy for beginners to contribute

**Reply by Sec-ant** [reply]
Can you be more specific? Or just post whatever the problem you met while setting up the repo?

**Comment by Md-Anamul-Haque** [comment]
We want to do Research and Development (R&D) on it.
For example, I want to add one more parameter to readBarcodesFromImageData which will be the third parameter of that function and it can be accessed from "src/cpp/ZXingWasm.cpp".

**Reply by Sec-ant** [reply, marked as answer]
If you want to contribute a feature to this repo, you should probably open an issue or discussion first, tell us details about the requirement, and then we can decide whether to adopt or reject it.

If you just want to add this feature in your own fork, you're more than welcome to do it as you wish, as long as it comply with the LICENSE.

As for how to implement the feature in the cpp file, well, they're basically standard cpp code, additionally using the embind API to communicate with the JavaScript code. I can't tell you how to write cpp code as I'm not skilled enough and I also don't have an answer, but if you want to learn more about the embind api, you can check its official document here: https://emscripten.org/docs/porting/connecting_cpp_and_javascript/embind.html Also google and chatgpt can be of great help.
